appservice/src/github/GitHubBranchListStep.ts

```diff
@@ -0,0 +1,33 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { AzureWizardPromptStep, IAzureQuickPickItem } from 'vscode-azureextensionui';
+import { ext } from '../extensionVariables';
+import { nonNullProp } from '../utils/nonNull';
+import { getGitHubQuickPicksWithLoadMore, gitHubBranchData, gitHubWebResource, ICachedQuickPicks } from './connectToGitHub';
+import { IConnectToGitHubWizardContext } from './IConnectToGitHubWizardContext';
+
+export class GitHubBranchListStep extends AzureWizardPromptStep<IConnectToGitHubWizardContext> {
+    public async prompt(context: IConnectToGitHubWizardContext): Promise<void> {
+        const placeHolder: string = 'Choose branch';
```

Comment:
"branchQuickPick" seems unnecessary and misleading since it's the data, not the pick

```
do {
    context.branchData = (await ext.ui.showQuickPick(this.getBranches(context, picksCache), { placeHolder })).data;
} while (!context.branchData);
```